checksum/checksum_test.go

```diff
@@ -0,0 +1,169 @@
+package checksum
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+	"time"
+)
+
+var testdata = filepath.Join("..", "testdata")
+
+func TestCreate(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		src      string
+		target   string
+		expected int
+		err      bool
+	}{
+		{
+			"Src directory does not exists",
+			"../no-dir",
+			t.TempDir(),
+			0,
+			true,
+		},
+		{
+			"Create checksum files",
+			testdata,
+			t.TempDir(),
+			16,
+			false,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, func(t *testing.T) {
+			err := Create(tc.src, tc.target)
+
+			if !tc.err && err != nil {
+				t.Error("unexpected err")
+			}
+
+			ls, err := os.ReadDir(tc.target)
+			if err != nil {
+				t.Errorf("expected no error reading dir %s, got %s", tc.target, err)
+			}
+
+			var got int
+			for _, f := range ls {
+				if !f.IsDir() {
+					got++
+				}
+			}
+
+			if got != tc.expected {
+				t.Errorf("expected %d files in the sample directory, got %d", tc.expected, got)
+			}
+		})
+	}
+}
+
+var missingfile = func(t *testing.T) string {
```

Comment:
```suggestion
var missingFile = func(t *testing.T) string {
```

E… pq não uma função?

```go
func missingFile(t *testing.T) string {
    …
}
```